libnavigation-base/src/main/java/com/mapbox/navigation/base/routerefresh/RouteOptionsProvider.kt

```diff
@@ -0,0 +1,24 @@
+package com.mapbox.navigation.base.routerefresh
+
+import android.location.Location
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.base.trip.model.RouteProgress
+
+/**
+ * Provider is used for *Reroute* and *Faster Route* flow.
+ *
+ * *Reroute* provider might be overwritten via `MapboxNavigation.setRouteOptionsProviderOnOffRoute`.
+ */
+interface RouteOptionsProvider {
+
+    /**
+     * Provide a new instance *RouteOptions* based on initial *RouteOptions*, *RouteProgress* and
+     * current *Location*
```

Comment:
nit: add documentation for how often this will be called. 

am wondering what would happen if you override this interface, and then return `null`. would that mean a new route progress will come in a second? (that would be good)

or does it cancel the reroute call?

Reply:
good point 👍

Reply:
Seems that route progress still will be called but with invalid state .I'm gonna create ticket to discuss it separately

Reply:
issue was created